tasks/system_probe.py

```diff
@@ -421,6 +421,7 @@ def test(
     windows=is_windows,
     failfast=False,
     kernel_release=None,
+    mod="mod",
```

Comment:
I'm curious what other `mod` value you needed?

Reply:
I used Go workspaces that were introduced in Go 1.18 when I made the payload changes. It allowed me to quickly iterate on the payload without having to push the payload & `go get` a different commit for every little fixes. But for them to work you need to use `mod=readonly`, which is already possible in `system-probe.build` with the `-g` option but not when running tests with `system-probe.test`. 

Now this change was staged in the middle of the huge merge of main ([601ba6d](https://github.com/DataDog/datadog-agent/pull/13516/commits/601ba6d8a89bcf8c1e3c1d72bc5c2ef8eed70b09)), and it is not really the point of this PR, so I can remove it, or make it another PR if you think this flag is a good addition.